src/zc/ffscript.cpp

```diff
@@ -42697,27 +42615,9 @@ void FFScript::getNPCData_misc()
 {
 	int32_t ID = int32_t(ri->d[rINDEX] / 10000); //the enemy ID value
 	int32_t indx = int32_t(ri->d[rINDEX2] / 10000); //the misc index ID
-	if ((ID < 1 || ID > 511) || ( indx < 0 || indx > 15 ))
+	if ((ID < 1 || ID > 511) || ( indx < 0 || indx > MAX_NPC_ATRIBUTES ))
 		set_register(sarg1, -10000); 
-	switch ( indx )
-	{
-		case 0: set_register(sarg1, guysbuf[ID].misc1 * 10000); break;
-		case 1: set_register(sarg1, guysbuf[ID].misc2 * 10000); break;
-		case 2: set_register(sarg1, guysbuf[ID].misc3 * 10000); break;
-		case 3: set_register(sarg1, guysbuf[ID].misc4 * 10000); break;
-		case 4: set_register(sarg1, guysbuf[ID].misc5 * 10000); break;
-		case 5: set_register(sarg1, guysbuf[ID].misc6 * 10000); break;
-		case 6: set_register(sarg1, guysbuf[ID].misc7 * 10000); break;
-		case 7: set_register(sarg1, guysbuf[ID].misc8 * 10000); break;
-		case 8: set_register(sarg1, guysbuf[ID].misc9 * 10000); break;
-		case 9: set_register(sarg1, guysbuf[ID].misc10 * 10000); break;
-		case 10: set_register(sarg1, guysbuf[ID].misc11 * 10000); break;
-		case 11: set_register(sarg1, guysbuf[ID].misc12 * 10000); break;
-		case 12: set_register(sarg1, guysbuf[ID].misc13 * 10000); break;
-		case 13: set_register(sarg1, guysbuf[ID].misc14 * 10000); break;
-		case 14: set_register(sarg1, guysbuf[ID].misc15 * 10000); break;
-		default: set_register(sarg1, -10000); break;
-	}
+	set_register(sarg1, guysbuf[ID].attributes[indx] * 10000);
```

Comment:
Out of bounds if this runs in the if case
```suggestion
	else set_register(sarg1, guysbuf[ID].attributes[indx] * 10000);
```